compiler/rustc_infer/src/infer/error_reporting/need_type_info.rs

```diff
@@ -392,14 +393,34 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
                     Applicability::HasPlaceholders,
                 );
             }
-            // FIXME: Consider autoref here. We sometimes
-            // have to add `&` or `&mut` in front of `self_span` here.
-            InferSourceKind::FullyQualifiedMethodCall { self_span, successor, substs, def_id } => {
+            InferSourceKind::FullyQualifiedMethodCall { receiver, successor, substs, def_id } => {
+                let typeck_results = self.in_progress_typeck_results.unwrap();
+                let typeck_results = typeck_results.borrow();
```

Comment:
I'd move all of these closer to the assignment of adjustment and make it an `if let` chain so that if `in_progress_typeck_results` happens to be `None`, we don't blow up. I don't know in what circumstance that would happen, but I've already caused ICEs that way.

Reply:
> I don't know in what circumstance that would happen, but I've already caused ICEs that way.

can't happen here, as we only add this infer source if there are typeck results. We don't have any typeck_results if the inference errors happens outside of a body, e.g. during coherence or whatever?

> I'd move all of these closer to the assignment of adjustment

do you want that regardless?

Reply:
The reason I want this is because this might be entirely safe today, but might not be after a refactor.